core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java

```diff
@@ -91,63 +92,67 @@ public static TTransportFactory transportFactory() {
   /**
    * Create a Thrift client using the given factory and transport
    */
-  public static <T extends TServiceClient> T createClient(TServiceClientFactory<T> factory,
+  public static <T extends TServiceClient> T createClient(ThriftClientType<T,?> type,
       TTransport transport) {
-    return factory.getClient(protocolFactory.getProtocol(transport),
-        protocolFactory.getProtocol(transport));
+
+    TProtocol protocol = protocolFactory.getProtocol(transport);
+    if (type.isMultiplexed()) {
```

Comment:
I think this can be simplified slightly if we just make everything multiplexed (since it's basically no cost, and can help make rpc extensible in future).

Reply:
I made everything multiplexed in 2bac8d5f30828ada41d60168272b0c6b46802dc5